### DISCUSSION TITLE
Error in download_jira_data.Rmd

### DISCUSSION THREAD

**Discussion by ian-lastname** [sailuh/kaiaulu#264] [Q&A] [answered]
On line 96:
jira_issues_list <- parse_jira(save_path_issue_tracker_issues)

I get this warning:
"Warning: cannot open file '../.../rawdata/issue_tracker/geronimo_issues.json' : No such file or directoryError in open.connection(con, "rb") : cannot open the connection"

**Comment by carlosparadis** [comment]
Is the file it is looking for in that location? The downloader should create this file if the download succeeds.

**Reply by ian-lastname** [reply, marked as answer]
We found out that the folder organization for rawdata is not being used. We also found out that the require(jiragiler) was not working but the error message was not showing because the code block had message = false warning = false. Before that we found that the file could not be opened because the function that downloads it did not exist and because the folder that was supposed to be saved also didn't exist.

**Reply by Ssunoo2** [reply]
Did you get this issue to work? I was stuck at the same spot that you were. I likewise did not have JirAgileR error messages pop up. I used devtools::install_github("matbmeijer/JirAgileR") to install the package but am still having the same issue.

**Reply by ian-lastname** [reply]
Did you set up the directory correctly? Turns out that we have to create the file paths that the function is trying to put the file in.

**Reply by carlosparadis** [reply]
@Ssunoo2 As @ian-lastname noted, there are two parts to getting this to work: The first is the dev tools to manually install the package, which it seems you already did. The second is the folder structure you should have surrounding Kaiaulu so the data file is downloaded.

To go with the default assumed in the Notebook, and assuming the code you downloaded is called `kaiaulu-master`, then:

 * /anywhere/in/your/computer/kaiaulu/kaiaulu-master
 * /anywhere/in/your/computer/kaiaulu/rawdata/issue_tracker

the rest of the sub-folder structure should be created by the Notebook if needed. 

For the record, this problem you are experiencing relates to:

 * https://github.com/sailuh/kaiaulu/issues/257
 * https://github.com/sailuh/kaiaulu/issues/252

**Reply by Ssunoo2** [reply]
So i set up the paths correctly. I actually just changed them in the config file to test. I still get the same error when only the directories are set up. Then I created the two files geronimo_issue_comments.json and geronimo_issues to see if they could be read with or without actual data and got a new error on line 96

https://github.com/sailuh/kaiaulu/blob/9294a9bb92506b76424b0fcc7c07f6b74d5ce53d/vignettes/download_jira_data.Rmd#L96

Error in parse_con(txt, bigint_as_char) : parse error: premature EOF
                                       
                     (right here) ------^

I checked the two .json files in VSCO and they are still empty so somehow the data is not getting downloaded I am assuming. I confirmed again that I do have JirAgileR installed by rerunning devtools::install_github("matbmeijer/JirAgileR"), which returned: 

Skipping install of 'JirAgileR' from a github remote, the SHA1 (7626a419) has not changed since last install.

I'm still not sure why I am getting errors.

**Reply by Ssunoo2** [reply]
I found out that I had to run the functions one by one and that using "run all" would result in no data being downloaded
